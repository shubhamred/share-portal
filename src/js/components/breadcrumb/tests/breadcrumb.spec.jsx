import React from 'react';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Breadcrumbs from '../breadcrumb';

configure({ adapter: new Adapter() });

describe('components/breadcrumb', () => {
  it('Renders default options', () => {
    const component = shallow(<Breadcrumbs />);
    const tree = shallowToJson(component);
    expect(tree).toMatchSnapshot();
  });

  describe('Check for click handle', () => {
    const props = {
      BreadcrumbsArray: []
    };
    // eslint-disable-next-line react/jsx-props-no-spreading
    const ButtonComponentWrapper = shallow(<Breadcrumbs {...props} />);

    it('check prop type children should not be empty', () => {
      expect(ButtonComponentWrapper.props().children).toBeTruthy();
    });
  });
});
