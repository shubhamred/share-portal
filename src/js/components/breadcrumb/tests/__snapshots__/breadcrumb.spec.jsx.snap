// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/breadcrumb Renders default options 1`] = `
<WithStyles(ForwardRef(Grid))
  className="makeStyles-formContainer-1"
  container={true}
  direction="row"
  item={true}
  justify="space-between"
  style={Object {}}
>
  <div
    className="makeStyles-divBreadcrumbs-2"
  >
    <WithStyles(ForwardRef(Breadcrumbs))
      aria-label="breadcrumb"
    />
  </div>
</WithStyles(ForwardRef(Grid))>
`;
